modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/RegionResourceImpl.java

```diff
@@ -111,22 +116,52 @@ public Page<Region> getRegionsPage(
 	public Region postCountryRegion(Long countryId, Region region)
 		throws Exception {
 
-		return _toRegion(
+		if (region.getTitle_i18n() == null) {
+			Map<String, String> titleMap = new HashMap<>();
+
+			for (Locale locale : _language.getAvailableLocales()) {
+				titleMap.put(_language.getLanguageId(locale), null);
+			}
+
+			region.setTitle_i18n(titleMap);
+		}
+
+		com.liferay.portal.kernel.model.Region serviceBuilderRegion =
 			_regionService.addRegion(
 				countryId, GetterUtil.get(region.getActive(), true),
 				region.getName(), GetterUtil.getDouble(region.getPosition()),
 				region.getRegionCode(),
 				ServiceContextFactory.getInstance(
-					Region.class.getName(), contextHttpServletRequest)));
+					Region.class.getName(), contextHttpServletRequest));
+
+		_regionLocalService.updateRegionLocalizations(
```

Comment:
Is it correct to use the regionLocalService? Because on row 130 we are using the regionService?

Reply:
@smottal yeah, since updateRegionLocalizations is only in RegionLocalService. The only difference would be the permission checking. In this case it would check for the same permission, MANAGE_COUNTRIES. The regionService call before the regrionLocalService call will do the permission check.